File: mobile/lib/models/subscription.dart
class Subscription {
  final String id;
  final String userId;
  final String plan; // 'monthly' or 'quarterly'
  final double amount;
  final DateTime startDate;
  final DateTime endDate;
  final String status; // 'active', 'expired', 'cancelled'
  final String? mpesaReceiptNumber;
  final DateTime createdAt;

  Subscription({
    required this.id,
    required this.userId,
    required this.plan,
    required this.amount,
    required this.startDate,
    required this.endDate,
    required this.status,
    this.mpesaReceiptNumber,
    required this.createdAt,
  });

  bool get isActive {
    return status == 'active' && DateTime.now().isBefore(endDate);
  }

  factory Subscription.fromMap(Map<String, dynamic> data, String id) {
    return Subscription(
      id: id,
      userId: data['userId'] ?? '',
      plan: data['plan'] ?? 'monthly',
      amount: (data['amount'] ?? 0).toDouble(),
      startDate: DateTime.parse(data['startDate'] ?? DateTime.now().toIso8601String()),
      endDate: DateTime.parse(data['endDate'] ?? DateTime.now().toIso8601String()),
      status: data['status'] ?? 'active',
      mpesaReceiptNumber: data['mpesaReceiptNumber'],
      createdAt: DateTime.parse(data['createdAt'] ?? DateTime.now().toIso8601String()),
    );
  }

  Map<String, dynamic> toMap() {
    return {
      'userId': userId,
      'plan': plan,
      'amount': amount,
      'startDate': startDate.toIso8601String(),
      'endDate': endDate.toIso8601String(),
      'status': status,
      'mpesaReceiptNumber': mpesaReceiptNumber,
      'createdAt': createdAt.toIso8601String(),
    };
  }
}

File: backend/models/database/Subscription.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const Subscription = sequelize.define('Subscription', {
  id: {
    type: DataTypes.UUID,
    defaultValue: DataTypes.UUIDV4,
    primaryKey: true
  },
  userId: {
    type: DataTypes.UUID,
    allowNull: false
  },
  plan: {
    type: DataTypes.STRING,
    defaultValue: 'basic',
    enum: ['basic', 'premium', 'annual']
  },
  status: {
    type: DataTypes.STRING,
    defaultValue: 'active',
    enum: ['active', 'cancelled', 'expired']
  },
  startDate: {
    type: DataTypes.DATE,
    defaultValue: DataTypes.NOW
  },
  expiryDate: {
    type: DataTypes.DATE,
    allowNull: false
  },
  amount: {
    type: DataTypes.DECIMAL(10, 2),
    allowNull: false
  },
  createdAt: {
    type: DataTypes.DATE,
    defaultValue: DataTypes.NOW
  },
  updatedAt: {
    type: DataTypes.DATE,
    defaultValue: DataTypes.NOW
  }
}, {
  tableName: 'subscriptions',
  timestamps: true
});

module.exports = Subscription;
